app/lib/features/settings/pages/calendar_settings_page.dart

```diff
@@ -37,18 +39,39 @@ class CalendarSettingsPage extends ConsumerWidget {
                           true),
                   onToggle: (newVal) async {
                     if (newVal) {
-                      final hasPermission = await handleCalendarPermission( context, );
-                      if (!hasPermission || !context.mounted) {
-                        return;
+                      final hasPermission = await isShowCalendarPermissionInfoPage();
```

Comment:
this is a pretty complex change that is hard for me to follow as the names don't really make sense to me ... how does `isShowCalendarPermissionInfoPage` become `hasPermission` and why then show a dialog, it already has the permissions?

This is confusing... and not covered by tests. Makes me a bit uneasy about it.